prov/sockets/src/sock_fabric.c

```diff
@@ -382,6 +384,9 @@ SOCKETS_INI
 	fi_param_define(&sock_prov, "iface", FI_PARAM_STRING,
 			"Specify interface name");
 
+	fi_param_define(&sock_prov, "def_buf_sz", FI_PARAM_INT,
+                        "Default send/recv buffer size");
```

Comment:
Please update message to be more descriptive, and consider renaming from default to max.  e.g. "Maximum total per socket send and recv buffer sizes (i.e. SO_RCVBUF, SO_SNDBUF)"

Reply:
I can improve the message. However I think "default" is correct.  Linux has three numbers for buffer size. for example:

> cat /proc/sys/net/ipv4/tcp_rmem
16384	349520	16777216

They are min, default and max. What we are changing here is the second number (default).

For this PR, I think I can remove "def" to avoid the confusion.

Reply:
The value is being fed into SO_RCVBUF and SO_SNDBUF, which are maximum values according to the man pages.  E.g.

"SO_SNDBUF
              Sets or gets the maximum socket send buffer in bytes."

https://man7.org/linux/man-pages/man7/socket.7.html

Reply:
yes, that's why I said it is confusing. If reading further, it says:

              The default value is set by the
              /proc/sys/net/core/rmem_default file, and the maximum
              allowed value is set by the /proc/sys/net/core/rmem_max
              file.  The minimum (doubled) value for this option is 256.

These are the three numbers (min, default and max). The default is what this function is overriding. 
But I can go with "max" if you think it is better.

Reply:
done

Reply:
Conceptually, I believe it is actually setting the maximum size of buffer space allowed for this socket, subject to system constraints.  That is, whatever value is set is the largest amount of buffer space that will be dedicated to the socket.  There's no guarantee that this limit will be reached however.  This can be used for scaling purposes.

By default, the maximum size of the buffer space is set to rmem_default.  An app may override that default, up to rmem_max.  My understanding is that the true 'default buffer space' allocated for a socket is basically just a single mtu.  It increases above that as needed, up to the SO_RCVBUF limit.  So if a connection is established that never receives data it will consume minimal memory.

In any case, referencing the actual socket options that it controls is likely sufficient documentation.  :)